### PR TITLE
support fit spark dataframe in keras model

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 1]
```diff
@@ -472,6 +473,70 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
...
+
+    this.fit(trainingData, nbEpoch, valData)
+  }
+
+  def fit(
+      x: DataFrame,
+      batchSize: Int,
+      nbEpoch: Int,
+      featureCol: String,
+      labelCol: String)(implicit ev: TensorNumeric[T]): Unit = {
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:536]: both feature and label columns should be a list of string, similar to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/src/bigdl/orca/learn/pytorch/pytorch_spark_estimator.py#L104

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 2]
```diff
@@ -472,6 +473,70 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
...
+    } else {
+      (row: Row, index: Int) => row.get(index)
+    }
+  }
+
+  private def getDataSet(
+      dataFrame: DataFrame,
+      batchSize: Int,
+      featureCol: String,
+      labelCol: String): FeatureSet[MiniBatch[T]] = {
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:489]: featureCol and labelCol should be a list of strings, similar to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/src/bigdl/orca/learn/pytorch/pytorch_spark_estimator.py#L68

**Comment by jason-dai** [comment]
Should align the APIs for `keras.model` and `FLModel` (https://github.com/intel-analytics/BigDL/pull/3685), and refactor the method of Dataframe to DataSet into one. @Litchilitchy please take a look.

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala (hunk 2 above)
Comment (Litchilitchy) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:489]: In scala these two are `Array[String]` currently

**Comment by dding3** [comment]
jekins pass. http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-PR-Dllib-Validation/145/

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 3]
```diff
@@ -485,24 +486,35 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
   private def getDataSet(
       dataFrame: DataFrame,
       batchSize: Int,
-      featureCol: String,
+      featureCols: Array[String],
       labelCol: String): FeatureSet[MiniBatch[T]] = {
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:490]: `labelCol` also needs to be a list or array of string

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 4]
```diff
@@ -472,6 +474,82 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
...
+  }
+
+  def fit(
+      x: DataFrame,
+      batchSize: Int,
+      nbEpoch: Int,
+      featureCols: Array[String],
+      labelCol: Array[String],
+      valX: DataFrame)(implicit ev: TensorNumeric[T]): Unit = {
+    require(labelCol.size == 1, "currently only suport set one column label")
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:535]: why only one column label?

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala [hunk 5]
```diff
@@ -472,6 +474,82 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
...
+    } else {
+      dataFrame.withColumnRenamed(featureCols.head, "features")
+    }
+
+    val featureCol = "features"
+    val featureColIndex = df.schema.fieldIndex(featureCol)
+    val featureType = df.schema(featureCol).dataType
+    val featureFunc = unwrapVectorAsNecessary(featureType)
+
+    val labelFunc: (Row) => Option[Any] = if (df.columns.contains(labelCol)) {
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala:509]: what does it mean if `labelCol` is not in the dataframe?
